tests/ui/const-generics/generic_const_parameter_types/bad_inference.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(
+    adt_const_params,
+    unsized_const_params,
+    generic_const_parameter_types,
+    generic_arg_infer
+)]
+#![allow(incomplete_features)]
+
+use std::marker::ConstParamTy_;
+
+fn foo<T: ConstParamTy_, const N: usize, const M: [T; N]>() -> [T; N] {
+    loop {}
+}
+
+fn main() {
+    // Requires inferring `T`/`N` from `12_u8` and `2` respectively.
+    let a = foo::<_, _, { [12_u8; 2] }>();
```

Comment:
hm, is it different for these cases?
```rust
fn main() {
    let a = foo::<_, _, { [5_u8, 6] }>();
    let b: [u8; 2] = foo::<_, _, { [5, 6] }>();
}
```

Reply:
those would both error, the requirement is that from looking at the generic arguments to `foo` you must be able to determine the full inferred type of the const parameter without performing any type inference or trait solving. (though this is only the case when the argument is an anonymous constant, as opposed to a `_` or a usage of a generic parameter or some mgca arg)